# Практическая работа №2: Обработка выборочных данных. Нахождение точечных оценок параметров распределения

Выполнил студент гр. 1381 Богданов Егор. Вариант №1

## Цель работы

Получение практических навыков нахождения точечных статистических оценок параметров распределения.

## Основные теоретические положения
Начальный эмпирический момент k-ого порядка: $$\overline M_k=\frac{\sum{n_jx^k_j}}{N}$$

Центральный эмпирический момент:
$$\overline m_k=\frac{\sum{n_j(x_j-\overline x_v)^k}}{N}$$

Выборочная дисперсия:
$$D_v=\overline m_2=\frac{\sum{n_j(x_j-\overline x_v)^2}}{N}$$

Несмещённая выборочная дисперсия:
$$s^2=\frac{ND_v}{N-1}$$

Среднеквадратичное отклонение:
$$\sigma_v=\sqrt{D_v} \quad s=\sqrt{s^2}$$

Ассиметрия:
$$\overline A_s=\frac{\overline m_3}{s^3}$$

Эксцесс:
$$\overline E=\frac{\overline m_4}{s^4}-3$$

Условная варианта:
$$ u_j=\frac{x_j-C}{h}$$

Условный момент K-ого порядка:
$$\overline M^*_k=\frac{1}{N} \sum{n_j(\frac{x_j-C}{h})^k}=\frac{1}{N} \sum{n_ju^k_j}$$

Связь условных эмпирических моментов с эмпирическими начальными и центральными:
$$\overline M_1=\overline M^*_1h+C$$
$$\overline m_2=(\overline M^*_2-(\overline M^*_1)^2)h^2$$
$$\overline m_3=(\overline M_3^* -3\overline M_2^* + 2(\overline M_1^*)^3)h^3$$
$$\overline m_4=(\overline M^*_4 -4\overline M_3^*\overline M_1^*+\overline M^*_2(\overline M^*_1)^2-3(\overline M^*_1)^4)h^4$$

Контрольная сумма:
$$\sum{n_j(u_j+1)^4}=\sum{n_ju_j^4}+4\sum{n_ju_j^3}+6\sum{h_ju_j^2}+4\sum{n_ju_j}+1$$

Мода:
$$M_o^*=x_{i-1}+h\frac{\tilde m_i-\tilde m_{i-1}}{(\tilde m_i-\tilde m_{i-1})+(\tilde m_i - \tilde m_{i+1})}$$
где i-номер интервала с наибольшей частотой

Медиана:
$$M_e^*=x_i+\frac{h}{\tilde m_i}(0.5-\sum^{i-1}_{j=0}{\tilde m_j })$$

Выборочный коффициент вариации:
$$V^*=\frac{\sigma_v}{\overline x_v}*100\%$$

## Постановка задачи

Для заданных выборочных данных вычислить с использованием метода моментов и условных вариант точечные статистические оценки математического ожидания, дисперсии, среднеквадратичного отклонения, асимметрии, эксцесса, моды, медианы и коэффициента вариации исследуемой случайной величины. Полученные результаты содержательно проинтерпретировать.

Порядок выполнения работы:
1. Для середин интервального ряда, полученного в практической работе №1, вычислить условные варианты. Заполнить табл. 1 (в последней строке Σ необходимо заполнить суммы столбцов; ячейки отмеченные прочерком заполнять не надо). Провести контроль вычислений.
2. Вычислить условные эмпирические моменты $\overline M^∗_k$ через условные варианты. С помощью условных эмпирических моментов вычислить центральные эмпирические моменты $\overline m_k$. Полученные результаты занести в табл. 2.
3. Вычислить выборочные среднее $\overline x_v$ и дисперсию $D_v$
 с помощью стандартной формулы и с помощью условных вариант. Убедиться, что результаты совпадают. Вычислить выборочное СКО $σ_в$.
4. Вычислить исправленную выборочную дисперсию $s^2$ и исправленное СКО s. Сравнить данные оценки с смещёнными оценками дисперсии и СКО. Сделать выводы.
5. Найти статистическую оценку коэффициентов асимметрии $\overline A_s$ и эксцесса $\overline E$. Сделать выводы.

6. Для интервального ряда вычислить моду $M^∗_o$, медиану $M^∗_e$ и коэффициент вариации $V^∗$ заданного распределения. Сделать выводы.

## Выполнение работы
### Задание 1


In [104]:
N=97
import pandas as pd
import numpy as np
import math 
file=pd.read_csv('data.csv', delimiter=';', encoding='utf-8')
X=file['nu']
Y=file['E']
Xr=X.sort_values().reset_index(drop=True)
Yr=Y.sort_values().reset_index(drop=True)
Xv=Xr.drop_duplicates(keep='first').reset_index(drop=True)
Yv=Yr.drop_duplicates(keep='first').reset_index(drop=True)


ky=math.floor(1+3.31*math.log(Yv.size,10))
kx=math.floor(1+3.31*math.log(Xv.size,10))

hx=(Xv.max()-Xv.min())/kx
hy=(Yv.max()-Yv.min())/ky

Intervals_X=[]
for i in range(1,kx+1):
    Intervals_X.append([float(Xv.min()+(i-1)*hx),float(Xv.min()+i*hx)])
    
Intervals_Y=[]

for i in range(1,ky+1):
    Intervals_Y.append([float(Yv.min()+(i-1)*hy),float(Yv.min()+i*hy)])


Xtilda=[]
Ytilda=[]
for i in range(7):
    Xtilda.append((Intervals_X[i][0]+Intervals_X[i][1])/2)
    Ytilda.append((Intervals_Y[i][0]+Intervals_Y[i][1])/2)

My=[0,0,0,0,0,0,0]
Mx=[0,0,0,0,0,0,0]
print('\n')
for i in range(6):
    for j in range(Xv.size):
        if Intervals_X[i][0]<=Xv[j] and Intervals_X[i][1]>Xv[j]:
            Mx[i]+=1
    for j in range(Yv.size):
        if Intervals_Y[i][0]<=Yv[j] and Intervals_Y[i][1]>Yv[j]:
            My[i]+=1

for j in range(Xv.size):
        if Intervals_X[6][0]<=Xv[j] and Intervals_X[6][1]>=Xv[j]:
            Mx[6]+=1
for j in range(Yv.size):
        if Intervals_Y[6][0]<=Yv[j] and Intervals_Y[6][1]>=Yv[j]:
            My[6]+=1

Nx=[]
Ny=[]

for i in range(7):
    Nx.append(Mx[i]/Xv.size)
    Ny.append(My[i]/Yv.size)


Ux=[]
Uy=[]
Cx=Xtilda[3]
Cy=Ytilda[3]

for i in range(0,7):
    Ux.append((Xtilda[i]-Cx)/hx)
    Uy.append((Ytilda[i]-Cy)/hy)

print("Ux=",Ux)
print("Uy=",Uy)



Ux= [np.float64(-3.000000000000001), np.float64(-1.9999999999999996), np.float64(-0.9999999999999998), np.float64(0.0), np.float64(0.9999999999999983), np.float64(2.0000000000000027), np.float64(3.000000000000001)]
Uy= [np.float64(-2.9999999999999987), np.float64(-1.9999999999999991), np.float64(-0.9999999999999996), np.float64(0.0), np.float64(1.0000000000000013), np.float64(2.000000000000001), np.float64(3.0000000000000004)]


Нецелый результат является следствием неточности работы с форматом плавающей точки. Для удобства значения будут округлены до целых

In [105]:
Ux=[]
Uy=[]
for i in range(0,7):
    Ux.append(round((Xtilda[i]-Cx)/hx))
    Uy.append(round((Ytilda[i]-Cy)/hy))

print("Ue=",Ux)
print("Unu=",Uy)

Nx=np.array(Nx)
Ny=np.array(Ny)
Ux=np.array(Ux)
Uy=np.array(Uy)

NUx=Nx*Ux
NU2x=Nx*Ux*Ux
NU3x=Nx*Ux*Ux*Ux
NU4x=Nx*Ux*Ux*Ux*Ux
NU41x=Nx*(Ux+1)*(Ux+1)*(Ux+1)*(Ux+1)
print("Nx*Ux=",NUx)
print("Nx*Ux^2=",NU2x)
print("Nx*Ux^3=",NU3x)
print("Nx*Ux^4=",NU4x)
print("Nx*(Ux+1)^4=",NU41x)

print("sum Nx*Ux=",np.sum(NUx))
print("sum Nx*Ux^2=",np.sum(NU2x))
print("sum Nx*Ux^3=",np.sum(NU3x))
print("sum Nx*Ux^4=",np.sum(NU4x))
print("sum Nx*(Ux+1)^4=",np.sum(NU41x))

print("check:",np.sum(NU4x)+4*np.sum(NU3x)+6*np.sum(NU2x)+4*np.sum(NUx)+1-np.sum(NU41x))

Ue= [-3, -2, -1, 0, 1, 2, 3]
Unu= [-3, -2, -1, 0, 1, 2, 3]
Nx*Ux= [-0.4109589  -0.43835616 -0.31506849  0.          0.1369863   0.08219178
  0.08219178]
Nx*Ux^2= [1.23287671 0.87671233 0.31506849 0.         0.1369863  0.16438356
 0.24657534]
Nx*Ux^3= [-3.69863014 -1.75342466 -0.31506849  0.          0.1369863   0.32876712
  0.73972603]
Nx*Ux^4= [11.09589041  3.50684932  0.31506849  0.          0.1369863   0.65753425
  2.21917808]
Nx*(Ux+1)^4= [2.19178082 0.21917808 0.         0.12328767 2.19178082 3.32876712
 7.01369863]
sum Nx*Ux= -0.8630136986301371
sum Nx*Ux^2= 2.9726027397260273
sum Nx*Ux^3= -4.561643835616438
sum Nx*Ux^4= 17.93150684931507
sum Nx*(Ux+1)^4= 15.068493150684931
check: -1.7763568394002505e-15


Таблица 1 для столбца nu:
| i       |$\tilde{x_i}$    | $n_i$            |$u_i$  | $n_iu_i$  |  $n_iu^2_i$|$n_iu^3_i$ |$n_iu^4_i$|$n_i(u_i+1)^4$|
|--------|------------------|------------------|-------|--------   |------------|---------- |----------|----------|
| 1      |380.6428571428571 |0.136986301369863 |  -3   |-0.4109589 |1.23287671  |-3.69863014|11.09589041|2.19178082|
| 2      |417.92857142857144|0.2191780821917808|  -2   |-0.43835616|0.87671233  |-1.75342466|3.50684932|0.21917808 |
| 3      |455.2142857142857 |0.3150684931506849|  -1   |-0.31506849|0.31506849  |-0.31506849|0.31506849|0          |
| 4      |492.5             |0.1232876712328767|   0   |0          |0           |0          |0         |0.12328767|
| 5      |529.7857142857142 |0.136986301369863 |   1   |0.1369863  |0.1369863   |0.1369863  |0.1369863 |2.19178082|
| 6      |567.0714285714287 |0.0410958904109589|   2   |0.08219178 |0.16438356  |0.32876712 |0.65753425|3.32876712|
| 7      |604.3571428571429 |0.0273972602739726|   3   |0.08219178 |0.24657534  |2.21917808 |2.21917808|7.01369863|
| $\sum$ |-                 |1                 |     - |-0.8630136986301371           |2.9726027397260273            |-4.561643835616438           | 17.93150684931507         |  15.068493150684931         |

In [106]:
NUy=Ny*Uy
NU2y=Ny*Uy*Uy
NU3y=Ny*Uy*Uy*Uy
NU4y=Ny*Uy*Uy*Uy*Uy
NU41y=Ny*(Uy+1)*(Uy+1)*(Uy+1)*(Uy+1)
print("Ny*Uy=",NUy)
print("Ny*Ux^2=",NU2y)
print("Ny*Ux^3=",NU3y)
print("Ny*Ux^4=",NU4y)
print("Ny*(Uy+1)^4=",NU41y)

print("sum Ny*Uy=",np.sum(NUy))
print("sum Ny*Uy^2=",np.sum(NU2y))
print("sum Ny*Uy^3=",np.sum(NU3y))
print("sum Ny*Uy^4=",np.sum(NU4y))
print("sum Ny*(Uy+1)^4=",np.sum(NU41y))
print("check:",np.sum(NU4y)+4*np.sum(NU3y)+6*np.sum(NU2y)+4*np.sum(NUy)+1-np.sum(NU41y))

Ny*Uy= [-0.12903226 -0.30107527 -0.38709677  0.          0.13978495  0.06451613
  0.09677419]
Ny*Ux^2= [0.38709677 0.60215054 0.38709677 0.         0.13978495 0.12903226
 0.29032258]
Ny*Ux^3= [-1.16129032 -1.20430108 -0.38709677  0.          0.13978495  0.25806452
  0.87096774]
Ny*Ux^4= [3.48387097 2.40860215 0.38709677 0.         0.13978495 0.51612903
 2.61290323]
Ny*(Uy+1)^4= [0.68817204 0.15053763 0.         0.21505376 2.23655914 2.61290323
 8.25806452]
sum Ny*Uy= -0.5161290322580644
sum Ny*Uy^2= 1.9354838709677418
sum Ny*Uy^3= -1.4838709677419353
sum Ny*Uy^4= 9.548387096774192
sum Ny*(Uy+1)^4= 14.161290322580644
check: -1.7763568394002505e-15


Таблица 1 для столбца e:
| i       |$\tilde{x_i}$    | $n_i$            |$u_i$  | $n_iu_i$  |  $n_iu^2_i$|$n_iu^3_i$ |$n_iu^4_i$|$n_i(u_i+1)^4$|
|--------|------------------|------------------|-------|--------   |------------|---------- |----------|----------|
| 1      |380.6428571428571 |0.136986301369863 |  -3   |-0.12903226|0.38709677  |-1.16129032|3.48387097|0.68817204|
| 2      |417.92857142857144|0.2191780821917808|  -2   |-0.30107527|0.60215054  |-1.20430108|2.40860215|0.15053763 |
| 3      |455.2142857142857 |0.3150684931506849|  -1   |-0.38709677|0.38709677  |-0.38709677|0.38709677|0          |
| 4      |492.5             |0.1232876712328767|   0   |0          |0           |0          |0         |0.21505376|
| 5      |529.7857142857142 |0.136986301369863 |   1   |0.13978495 |0.13978495  |0.13978495 |0.13978495 |2.23655914|
| 6      |567.0714285714287 |0.0410958904109589|   2   |0.06451613 |0.25806452  |0.25806452 |0.51612903|2.61290323|
| 7      |604.3571428571429 |0.0273972602739726|   3   |0.09677419 |0.87096774  |0.87096774 |2.61290323|8.25806452|
| $\sum$ |-                 |1                 |     - |-0.5161290322580644     |1.9354838709677418|-1.4838709677419353| 9.548387096774192|  14.161290322580644|

### Задание 2

In [107]:
M1x=np.sum(NUx)
M2x=np.sum(NU2x)
M3x=np.sum(NU3x)
M4x=np.sum(NU4x)

M1y=np.sum(NUy)
M2y=np.sum(NU2y)
M3y=np.sum(NU3y)
M4y=np.sum(NU4y)

m1x=0
m2x=(M2x-M1x**2)*(hx**2)
m3x=(M3x-3*M2x*M1x+2*M1x**3)*hx**3
m4x=(M4x-4*M3x*M1x+6*M2x*M1x**2-3*M1x**4)*hx**4

m1y=0
m2y=(M2y-M1y**2)*hy**2
m3y=(M3y-3*M2y*M1y+2*M1y**3)*hy**3
m4y=(M4y-4*M3y*M1y+6*M2y*M1y**2-3*M1y**4)*hy**4

print("Nu:")
print(M1x,m1x)
print(M2x,m2x)
print(M3x,m3x)
print(M4x,m4x)

print("E:")
print(M1y,m1y)
print(M2y,m2y)
print(M3y,m3y)
print(M4y,m4y)

Nu:
-0.8630136986301371 0
2.9726027397260273 3097.156153660563
-4.561643835616438 95844.60554642655
17.93150684931507 26679613.929215368
E:
-0.5161290322580644 0
1.9354838709677418 442.68478837945156
-1.4838709677419353 5347.487966751456
9.548387096774192 658812.1980410281


Таблица 2 для столбца nu:
| k      |$\overline{M^*_k}$ |$\overline{m^*_k}$ |
|--------|------------------ |------------------|
| 1      |-0.8630136986301371| 0                |
| 2      |2.9726027397260273 | 3097.156153660563|
| 3      |-4.561643835616438 |95844.60554642655 |
| 4      |17.93150684931507  |26679613.929215368|


Таблица 2 для столбца e:
| k      |$\overline{M^*_k}$ |$\overline{m^*_k}$ |
|--------|------------------ |------------------|
| 1      |-0.5161290322580644| 0                |
| 2      |1.9354838709677418 |442.68478837945156|
| 3      |-1.4838709677419353|5347.487966751456|
| 4      |9.548387096774192  |658812.1980410281|


### Задание 3

In [163]:
XlineS=np.sum(X)/N
YlineS=np.sum(Y)/N
XlineC=M1x*hx+Cx
YlineC=M1y*hy+Cy
print("Nu mean std=",XlineS," Nu mean cond=",XlineC)
print("E mean std=",YlineS," E mean cond=",YlineC)

Nu mean std= 458.5979381443299  Nu mean cond= 460.32191780821915
E mean std= 130.5371134020619  E mean cond= 130.29447004608295


In [162]:
DxS=np.sum((X-XlineS)**2)/N
DxC=m2x
DyS=np.sum((Y-YlineS)**2)/N
DyC=m2y

print("Dnu std=",DxS,"Dnu con=",DxC)
print("De std=",DyS,"De con=",DyC)

Dnu std= 2573.023913274524 Dnu con= 3097.156153660563
De std= 418.36790094590293 De con= 442.68478837945156


Неравенство результатов объясняется неточностью вычислений

In [160]:
sigmaX=math.sqrt(DxC)
sigmaY=math.sqrt(DyC)
print("Sigma nu=",sigmaX)
print("Sigma e=",sigmaY)

Sigma nu= 55.65209927451581
Sigma e= 21.04007576933723


### Задание 4

In [111]:
SX2=N/(N-1)*DxC
SX=math.sqrt(SX2)
print("Snu^2=",SX2,"Snu=",SX)

SY2=N/(N-1)*DyC
SY=math.sqrt(SY2)
print("Se^2=",SY2,"Se=",SY)

Sx^2= 3129.418196927861 Sx= 55.94120303432758
Sy^2= 447.2960882584042 Sy= 21.149375599728806


Из-за размера выборки разница между N и N-1 не является сильно ощутимой и поэтому результаты близки друг к другу

### Задание 5

In [158]:
Asyx=m3x/(SX**3)
Excx=m4x/(SX**4)-3
print("ASYnu=",Asyx,"EXCnu=",Excx)

Asyy=m3y/(SY**3)
Excy=m4y/(SY**4)-3
print("ASYe=",Asyy,"EXCe=",Excy)

ASYnu= 0.5474850556428541 EXCnu= -0.2757162892366303
ASYe= 0.5652715990482005 EXCe= 0.2928460852364694


Положительное значение ассиметрии показывает на правосторонний сдвиг

Положительное значение эксцесса указывает на увеличенную концентрацию данных у мат.ожидания. Отрицательное,соответственно,укзаывает на "сглаженную" вершину

### Задание 6

Мода $M^*_o$:

In [113]:
print(Nx)
print(Ny)

[0.1369863  0.21917808 0.31506849 0.12328767 0.1369863  0.04109589
 0.02739726]
[0.04301075 0.15053763 0.38709677 0.21505376 0.13978495 0.03225806
 0.03225806]


In [156]:

ix=2
Mox=Intervals_X[ix][0]+hx*((Nx[ix]-Nx[ix-1])/(Nx[ix]-Nx[ix-1]+Nx[ix]-Nx[ix+1]))
print("Mo nu=",Mox)

iy=2
Moy=Intervals_Y[iy][0]+hy*((Ny[iy]-Ny[iy-1])/(Ny[iy]-Ny[iy-1]+Ny[iy]-Ny[iy+1]))
print("Mo e=",Moy)

Monu= 449.0
Moe= 123.7


Медиана $M^*_e$:

In [157]:
mx=0
my=0
for i in range(2):
    mx+=Nx[i]
    my+=Ny[i]

Mex=Intervals_X[ix][0]+hx*(0.5-mx)/Nx[ix]
Mey=Intervals_Y[iy][0]+hy*(0.5-my)/Ny[iy]
print("Me nu=",Mex)
print("Me e=",Mey)

Menu= 453.59316770186336
Mee= 127.16428571428571


Коэффициент вариации 

In [154]:
print("Vnu=",sigmaX/XlineC*100)
print("Ve=",sigmaY/YlineC*100)

Vnu= 12.089821735949096
Ve= 16.148095741819056


Мода и медиана примерно равны,следовательно,распределение никуда не смещено

Коэффициент вариации демонстрирует разброс от среднего,но не очень большой

## Выводы

В ходе выполнения работы было сделано следующее:

1. Для середин интервального ряда, полученного в практической работе №1, вычислены условные варианты. Заполнена таблица. Проведён контроль вычислений.
2. Вычислены условные эмпирические моменты $\overline M^∗_k$ через условные варианты. С помощью условных эмпирических моментов вычислены центральные эмпирические моменты $\overline m_k$. Полученные результаты занесены в таблицу.
3. Вычислены выборочные среднее $\overline x_v$ и дисперсия $D_v$
 с помощью стандартной формулы и с помощью условных вариант. Убедиться, что результаты совпадают. Вычислено выборочное СКО $σ_в$.
4. Вычислена исправленная выборочная дисперсия $s^2$ и исправленное СКО s. Сравнены данные оценки с смещёнными оценками дисперсии и СКО. Сделаны выводы.
5. Найдена статистическая оценка коэффициентов асимметрии $\overline A_s$ и эксцесса $\overline E$. Сделаны выводы.

6. Для интервального ряда вычислена мода $M^∗_o$, медиана $M^∗_e$ и коэффициент вариации $V^∗$ заданного распределения. Сделаны выводы.